### DISCUSSION TITLE
May I ask how to modify the file system of MicroPython to FATFS

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12372] [ESP32]
When developing MicroPython, I found that the file system it uses does not seem to be FATFS.

Now I hope to use Freetype in the MicroPython environment, and Freetype must be FATFS when loading files, and I cannot modify the source code.

So, can the file system of MicroPython be switched to FATFS?

**Comment by robert-hh** [comment]
If the VFS includes the FAT driver, you can always reformat the file system.

```
import os
from flashbdev import bdev
os.VfsFat.mkfs(bdev)
vfs = os.VfsFat(bdev)
os.mount(vfs, "/")
```

**Reply by Mars-CN** [reply]
It seems that just setting up here is not enough. I just burned the bin file and used it

`esp_vfs_fat_spiflash_mount("/flash","vfs", &mnt_cfg, &handle);`

Loading partitions will fail, but if you switch to hello_ This is how to operate a blank project in the world.

So I suspect that MicroPython has already changed the partition format when burning because it needs to burn the boot. py file to the root directory.

So, are there any configurations that can be modified?

**Reply by karfas** [reply]
Don't suspect, look at the code.

*Flashing* micropython only creates a **partition**, no filesystem; the first boot after flashing creates the filesystem, using the functions in [esp32/modules](https://github.com/micropython/micropython/tree/master/ports/esp32/modules). 

I don't see a reason why @robert-hh 's code shouldn't work.

Maybe you need to reset your device immediate after ˋmkfs()ˋ?
And: Where is the connection to the esp-idf function you mentioned ?

**Reply by Josverl** [reply]
minor addition - in the default images there is no filesystem , but it definitely possible to add a formatted and even a populated vfs to an image.
I do this mostly with littlefs2 but is should work just the same with adding a properly* sized vsfFat partition image.

**Comment by jimmo** [comment]
@Mars-CN have you seen https://docs.micropython.org/en/latest/reference/filesystem.html#fat
